### ISSUE TITLE
[FAIL] HID USB device not ready

### ISSUE THREAD

**Issue by Klavyeli** [urbanadventurer/Android-PIN-Bruteforce#24]
Xiaomi mi9t nethunter 2021.1 
![IMG_20210421_062302](https://user-images.githubusercontent.com/38491552/115492222-0f307880-a26a-11eb-9b87-3cbe430098ab.jpg)

[Fail] HID USB device not ready. /system/xbin/hid-keyboard returned 127

how can we solve the problem my friend

**Comment by urbanadventurer** [comment]
Try unplugging and replugging the cables.

**Comment by rotatoria** [comment]
> Xiaomi mi9t nethunter 2021.1
> ![IMG_20210421_062302](https://user-images.githubusercontent.com/38491552/115492222-0f307880-a26a-11eb-9b87-3cbe430098ab.jpg)
> 
> [Fail] HID USB device not ready. /system/xbin/hid-keyboard returned 127
> 
> how can we solve the problem my friend

same problem here.

tried to restart devices. unplug and replug cables. nothing helped.

**Comment by Klavyeli** [comment]
The problem disappeared when I changed the cable on the otg cable, but it does not enter the screen even though there is the correct pin in the list and it does not mark the correct pin. Is there a config file for nexus 5 and mi9t?

**Comment by urbanadventurer** [comment]
I'm pleased to hear that it works with a new OTG cable.
@Klavyeli let's try to make a config file for your Nexus 5 and Mi9T.

Can you try plugging a real keyboard into your phone to discover the correct key, or combination of keys to use.
First, you need to find the keys that bring up the PIN prompt. This is defined in `KEYS_BEFORE_EACH_PIN`.
Next, you need to find what key sends the PIN. It is usually `enter` and defined in `KEYS_AFTER_EACH_PIN`.

You can send keys during the cooldown period to dismiss popups and other messages. This is defined in KEYS_STAY_AWAKE_DURING_COOLDOWN.

After you have the right keys, you'll need to figure out the timing.

**Comment by ante1377** [comment]
Have similar problem with my Nexus 5 Nethunter 2021 

[FAIL] HID USB device not ready. /system/xbin/hid-keyboard returned 5.

Tried reboot victim phone.
Tried different OTG cable.
Tried sending from Nexus 5 to Windows 10 CMD (working)
Tried swapping from MTP to Charge and back in Dev options (no change) 
Any ideas where to dig?

Victim is a HTC One S also tried on another HTC. Waiting for my other OTG dongels to try it on a Oneplus 5T.

**Comment by ante1377** [comment]
> The problem disappeared when I changed the cable on the otg cable, but it does not enter the screen even though there is the correct pin in the list and it does not mark the correct pin. Is there a config file for nexus 5 and mi9t?

Have you tried the DuckHunter HID from within Nethunter?

**Comment by urbanadventurer** [comment]
@ante1377 did you try connecting a USB keyboard to the OTG cable?

**Comment by ante1377** [comment]
> OTG cable?

Yes it works fine typing via a usb keyboard through the cable, think I only tried one of the three I bought to test with.

**Comment by ante1377** [comment]
Tried all these three just now and they all work. 
![IMG_20210426_175614](https://user-images.githubusercontent.com/83229710/116113647-f7695200-a6b8-11eb-8d85-757ae8ee99d1.jpg)

**Comment by ante1377** [comment]
adb logcat over :5555 does not display anything of interest that I can see.

**Comment by urbanadventurer** [comment]
@ante1377 please open a new issue for your phone

**Comment by Klavyeli** [comment]
> Yeni bir OTG kablosuyla çalıştığını duyduğuma sevindim.
> @Klavyeli Nexus 5 ve Mi9T'niz için bir yapılandırma dosyası oluşturmaya çalışalım.
> 
> Doğru tuşu veya kullanılacak tuş kombinasyonunu keşfetmek için telefonunuza gerçek bir klavye takmayı deneyebilir misiniz?
> Öncelikle, PIN istemini getiren anahtarları bulmanız gerekir. Bu, içinde tanımlanmıştır `KEYS_BEFORE_EACH_PIN`.
> Ardından, PIN'i hangi anahtarın gönderdiğini bulmanız gerekir. Genellikle `enter`ve tanımlanır `KEYS_AFTER_EACH_PIN`.
> 
> Açılır pencereleri ve diğer mesajları kapatmak için soğuma süresi sırasında anahtarlar gönderebilirsiniz. Bu, KEYS_STAY_AWAKE_DURING_COOLDOWN içinde tanımlanmıştır.
> 
> Doğru tuşlara sahip olduktan sonra, zamanlamayı bulmanız gerekecek.

Dude, how do you use it to see the correct key combination and keys that bring up the PIN prompt?

**Comment by Klavyeli** [comment]
@urbanadventurer Dude, how do you use it to see the correct key combination and keys that bring up the PIN prompt?

**Comment by urbanadventurer** [comment]
@Klavyeli to figure out the correct key combination you need to connect a real keyboard to the OTG cable and try different keys.

**Comment by urbanadventurer** [comment]
@Klavyeli Did you have any success in discovering a keyboard sequence to bring up the PIN prompt? Did you try combining keys with CTRL, SHIFT, ALT, etc?

**Comment by IamCOD3X** [comment]
Here is the solution https://youtu.be/OoMctGTSQ-A
